core/locks.go

```diff
@@ -1,78 +1,96 @@
 package core
 
-// LockOption represents a function that can modify a LockStrategy
-type LockOption func(*LockStrategy)
+import "log"
 
-// LockStrategy holds the state for our locking mechanism
-type LockStrategy struct {
-	storeLock    bool
-	storeRLock   bool
-	keypoolLock  bool
-	keypoolRLock bool
-}
+var (
+	LockHsh *LockHasher = NewLockHasher()
+)
+
+const (
+	// ReadLockOperation is used to acquire a read lock
+	ReadLockOperation = LockOperationT(1)
+	// WriteLockOperation is used to acquire a read lock
+	WriteLockOperation = LockOperationT(2)
+)
+
+type (
+	LockOperationT uint8
+
+	LockRequest struct {
+		Name LockName
+		Op   LockOperationT
+	}
+)
 
 // WithStoreLock sets the storeLock flag
-func WithStoreLock() LockOption {
-	return func(ls *LockStrategy) {
-		ls.storeLock = true
+func WithStoreLock() *LockRequest {
+	return &LockRequest{
+		Name: StoreLock,
+		Op:   ReadLockOperation,
```

Comment:
ReadLockOperation =>  WriteLockOperation

To follow same naming convention as that of LockOperationT should we rename WithStoreLock to WithStoreWriteLock or WithStoreReadWriteLock? wdyt?

Reply:
Done